indico/web/client/js/react/components/Combobox.jsx

```diff
@@ -0,0 +1,67 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React, {useRef} from 'react';
+
+import 'indico/custom_elements/ind_combobox';
+import {useNativeEvent} from 'indico/react/hooks';
+import {Translate} from 'indico/react/i18n';
+
+export default function Combobox({options, value, onChange, ...inputProps}) {
+  const uncontrolledInputProps = {...inputProps, defaultValue: value};
+  const inputRef = useRef();
+
+  useNativeEvent(inputRef, 'input', onChange);
+  useNativeEvent(inputRef, 'change', onChange);
+
+  return (
+    <ind-combobox>
+      <input ref={inputRef} {...uncontrolledInputProps} type="text" role="combobox" />
+      <ul role="listbox">
+        {options.map(function(valueLabel) {
+          let optionLabel, optionValue;
+          if (typeof valueLabel === 'string') {
+            optionLabel = valueLabel;
+            optionValue = valueLabel;
+          } else {
+            [optionValue, optionLabel] = valueLabel;
+          }
+
+          return (
+            <li role="option" data-value={optionValue} key={optionValue}>
+              {optionLabel}
+            </li>
+          );
+        })}
+      </ul>
+      <button type="button" value="clear" disabled={inputProps.disabled}>
```

Comment:
Is clicking this button supposed to open the dropdown? It feels strange because when I clear the field it typically means I want to to be empty, because in case I want to change the value I can simply open it again instead of clearing it first.